lib/matplotlib/artist.py

```diff
@@ -1421,6 +1421,10 @@ def setp(obj, *args, **kwargs):
       >>> setp(lines, 'linewidth', 2, 'color', 'r')  # MATLAB style
       >>> setp(lines, linewidth=2, color='r')        # python style
     """
+    
+    if ((isinstance(obj, np.ndarray) and not obj.size) or
```

Comment:
Agree using `if not objs:` would be better.  I would also move this down to L1453 (right above the line with `objs[0]`) after we have normalized the input to a list.